integrations/aws/aws/aws_credentials.py

```diff
@@ -9,18 +17,97 @@ def __init__(
         self.session_token = session_token
-        self.enabled_regions: list[str] = []
-        self.default_regions: list[str] = []
+        self.role_arn = role_arn
+        self.session_name = session_name
+        self.enabled_regions: List[str] = []
+        self.default_regions: List[str] = []
 
-    async def update_enabled_regions(self) -> None:
-        session = aioboto3.Session(
-            self.access_key_id, self.secret_access_key, self.session_token
+    async def _refresh_credentials(self, sts_client: STSClient) -> Dict[str, Any]:
+        """
+        Refreshes AWS credentials by re-assuming the role to get new credentials.
+
+        :return: A dictionary containing the new credentials and their expiration time.
+        """
+        response = await sts_client.assume_role(
+            RoleArn=str(self.role_arn),
+            RoleSessionName=str(self.session_name),
+            DurationSeconds=ASSUME_ROLE_DURATION_SECONDS,
         )
+        credentials = response["Credentials"]
+        self.access_key_id = credentials["AccessKeyId"]
+        self.secret_access_key = credentials["SecretAccessKey"]
+        self.session_token = credentials["SessionToken"]
+        expiry_time = credentials["Expiration"].isoformat()
+        return {
+            "access_key": self.access_key_id,
+            "secret_key": self.secret_access_key,
+            "token": self.session_token,
+            "expiry_time": expiry_time,
+        }
+
+    async def create_refreshable_session(
+        self, region: Optional[str] = None
+    ) -> aioboto3.Session:
+        """
+        Creates an aioboto3 Session with refreshable credentials.
+
+        :param region: AWS region for the session.
+        :return: An aioboto3 Session object.
+        """
+        if self.is_role():
+            session = aioboto3.Session(
+                aws_access_key_id=self.access_key_id,
+                aws_secret_access_key=self.secret_access_key,
+                aws_session_token=self.session_token,
+            )
+            async with session.client("sts") as sts_client:
+                initial_credentials = await self._refresh_credentials(sts_client)
+                refresh_credentials = partial(self._refresh_credentials, sts_client)
+                refreshable_credentials = (
+                    AioRefreshableCredentials.create_from_metadata(
+                        metadata=initial_credentials,
+                        refresh_using=refresh_credentials,
+                        method="sts-assume-role",
+                    )
+                )
+                botocore_session = get_session()
+                setattr(botocore_session, "_credentials", refreshable_credentials)
+                if region:
+                    botocore_session.set_config_variable("region", region)
+                autorefresh_session = aioboto3.Session(
+                    botocore_session=botocore_session
+                )
+                return autorefresh_session
```

Comment:
this is a pretty complex part of the code what do you think about adding maybe some comments about why this code exists and what is happening baisically?